# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.cpp
# Commit: 7ecc4f9b20ad
# Full Hash: 7ecc4f9b20ad70f95a63b96d8a53ebb36ee32d99
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 3
# Description:
#   Bug 1742437 - move DOM script context into own file and rename it; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138132
# ==============================================================================

diff -r 0b3f0972e002 -r 7ecc4f9b20ad dom/script/ScriptLoader.cpp
--- a/dom/script/ScriptLoader.cpp	Mon Feb 21 15:39:08 2022 +0000
+++ b/dom/script/ScriptLoader.cpp	Mon Feb 21 15:39:08 2022 +0000
@@ -623,8 +623,8 @@
     // this request be processed ahead of other Unblocked requests, but with
     // the same weight as Leaders.  This will make us behave similar way for
     // both http2 and http1.
-    DOMScriptLoadContext::PrioritizeAsPreload(channel);
-    DOMScriptLoadContext::AddLoadBackgroundFlag(channel);
+    ScriptLoadContext::PrioritizeAsPreload(channel);
+    ScriptLoadContext::AddLoadBackgroundFlag(channel);
   } else if (nsCOMPtr<nsIClassOfService> cos = do_QueryInterface(channel)) {
     if (aRequest->GetLoadContext()->mScriptFromHead &&
         aRequest->GetLoadContext()->IsBlockingScript()) {
@@ -777,7 +777,7 @@
   nsCOMPtr<Element> domElement = do_QueryInterface(aElement);
   RefPtr<ScriptFetchOptions> fetchOptions =
       new ScriptFetchOptions(aCORSMode, aReferrerPolicy, aTriggeringPrincipal);
-  RefPtr<DOMScriptLoadContext> context = new DOMScriptLoadContext(domElement);
+  RefPtr<ScriptLoadContext> context = new ScriptLoadContext(domElement);
 
   if (aKind == ScriptKind::eClassic) {
     RefPtr<ScriptLoadRequest> aRequest = new ScriptLoadRequest(